homeassistant/components/climate/generic_thermostat.py

```diff
@@ -217,4 +217,4 @@ def _control_heating(self):
     @property
     def _is_device_active(self):
         """If the toggleable device is currently active."""
-        return switch.is_on(self.hass, self.heater_entity_id)
+        return switch.async_is_on(self.hass, self.heater_entity_id)
```

Comment:
Figured out that core.is_state and core.is_state_attr are actually async friendly! So reverting that change.